src/ray/gcs/gcs_server/gcs_autoscaler_state_manager.cc

```diff
@@ -104,8 +98,25 @@ void GcsAutoscalerStateManager::HandleGetClusterStatus(
     rpc::autoscaler::GetClusterStatusRequest request,
     rpc::autoscaler::GetClusterStatusReply *reply,
     rpc::SendReplyCallback send_reply_callback) {
-  // TODO
-  throw std::runtime_error("Unimplemented");
+  auto ray_resource_state = reply->mutable_cluster_resource_state();
+  MakeClusterResourceStateInternal(ray_resource_state);
```

Comment:
Minor Q: what are differences between HandleGetClusterStatus and HandleGetClusterResourceState?

Reply:
- HandleGetClusterStatus: is what `ray status` calls for now
- HandleGetClusterResourceState: is what autoscaler will get

As of now, the 2 endpoints share many similarity. But we expect in the future, observability features (ray status) might contain more info than the one autoscaler gets.